src/components/DLP/DandisetOwnersDialog.vue

```diff
@@ -0,0 +1,302 @@
+                </span>
+              </span>
+
+              <span v-if="!isPrimaryOwner(owner)">
+                <v-btn
+                  text
+                  small
+                  @click="removeOwner(owner)"
+                >
+                  <v-icon
+                    color="error"
+                    left
+                  >mdi-minus-circle
+                  </v-icon>
+                  <span class="font-weight-medium">
+                    Remove
+                  </span>
+                </v-btn>
+                <v-btn
+                  text
+                  small
+                  @click="setPrimaryOwner(i)"
+                >
+                  <v-icon
+                    color="info"
+                    left
+                  >
+                    mdi-star-circle
+                  </v-icon>
+                  <span class="font-weight-medium">
+                    Make primary
+                  </span>
+                </v-btn>
+              </span>
+            </div>
+          </v-card>
+        </div>
+      </v-col>
+    </v-row>
+  </v-card>
+</template>
+
+<script lang="ts">
+import { debounce } from 'lodash';
+
+import { dandiRest } from '@/rest';
+import store from '@/store';
+import {
+  computed, defineComponent, Ref, ref, watch,
+} from '@vue/composition-api';
+import { User } from '@/types';
+
+export default defineComponent({
+  name: 'DandisetOwnersDialog',
+  props: {
+    owners: {
+      type: Array,
+      required: true,
+    },
+  },
+  setup() {
+    const currentDandiset = computed(() => store.state.dandiset.dandiset);
+    const owners = computed(() => store.state.dandiset.owners);
+
+    const searchQuery = ref('');
+    const loadingUsers = ref(false);
+    const newOwners: Ref<User[]> = ref([]);
+
+    // users with checkbox checked
+    const selectedUsers: Ref<User[]> = ref([]);
+
+    // eslint-disable-next-line no-underscore-dangle
+    const _searchResults: Ref<User[]> = ref([]);
+    const searchResults = computed(() => {
+      const newOwnersUsernames = new Set(newOwners.value.map((u: User) => u.username));
+      // only show non-owners in search results
+      return _searchResults.value.filter((u: User) => !newOwnersUsernames.has(u.username));
+    });
+
+    // Clear search results if search field is empty
+    watch(() => searchQuery.value, () => {
+      if (!searchQuery.value) {
+        _searchResults.value = [];
+      }
+    });
+
+    const throttledUpdate = debounce(async () => {
+      loadingUsers.value = true;
+      const users = await dandiRest.searchUsers(searchQuery.value);
+      _searchResults.value = users;
+      loadingUsers.value = false;
+    }, 200);
+
+    watch(() => owners.value, () => (
+      owners.value ? Object.assign(newOwners.value, owners.value) : null
+    ),
+    { immediate: true });
+
+    const isPrimaryOwner = (user: User) => newOwners.value?.length && newOwners.value[0] === user;
```

Comment:
What does being a primary owner entail? It looks like they get a special icon and they can't be deleted. Does that mean there's no way for the dandiset creator to abdicate ownership?

Reply:
My bad, I meant to put a comment explaining that. It's in Jared's design, but we don't support it yet. But, I think it might be a good idea.
Currently we don't have any notion of a "primary owner", but we do have a constraint that a dandiset must have at least one owner at all times. So right now, attempting to remove the last owner of a dandiset will fail. Maybe instead, it would be cleaner/better UX to have an explicitly designated "primary owner" that can never be removed. (any thoughts @waxlamp @satra @yarikoptic ?)

Reply:
i would for the moment keep it the way it is. i.e. owners have no differences in status. hence there is no concept of a primary owner, just a list of owners. any primariness is social within the group/project  that created the dandiset. i.e. it's just like our roles as admins on the archive.

Reply:
I agree with Satra; we shouldn't have the "primary owner" concept and treat all owners equally. This does mean we have to check at runtime and disallow removing a sole owner, but that also means transferring ownership is as simple as adding the transferree, then removing oneself from ownership.

Reply:
Sounds good, I removed the UI elements related to this